back/backend/lib/common.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import os
+
+from flask import request
+
+from backend.lib import formatters, auth, secrets
+from backend.models.region import Region
+from backend.models.province import Province
+from backend.models.user import Roles
+from backend.models.wca.event import Event
+from backend.models.wca.export import get_latest_export
+
+
+class Common(object):
+    current_date = datetime.datetime.now()
+
+    def __init__(self, wca_disclaimer=False):
+        self.uri = request.path
+        self.len = len
+        self.formatters = formatters
+        self.year = datetime.date.today().year
+        self.user = auth.user()
+        self.wca_disclaimer = wca_disclaimer
+
+        if self.user:
```

Comment:
I'd refactor this to
```
if not self.user:
    return
 
time_since_login = datetime.datetime.now() - self.user.last_login
self.just_logged_in = time_since_login < datetime.timedelta(seconds=1)
```

That way we avoid an additional level of indentation, and `self.just_logged_in` has a value of either `True` or `False`, rather than `True` or being undefined.